examples/developer_guide/2_2_rabbitmq/rabbitmq_source_stage.py

```diff
@@ -86,27 +86,29 @@ def name(self) -> str:
     def supports_cpp_node(self) -> bool:
         return False
 
+    def output_type(self) -> type:
+        return MessageMeta
+
     def stop(self):
         # Indicate we need to stop
         self._stop_requested = True
 
         return super().stop()
 
-    def _build_source(self, builder: mrc.Builder) -> StreamPair:
-        node = builder.make_source(self.unique_name, self.source_generator)
-        return node, MessageMeta
+    def _build_source(self, builder: mrc.Builder) -> mrc.SegmentObject:
+        return builder.make_source(self.unique_name, self.source_generator)
 
     def source_generator(self):
         try:
             while not self._stop_requested:
-                (method_frame, header_frame, body) = self._channel.basic_get(self._queue_name)
+                (method_frame, _, body) = self._channel.basic_get(self._queue_name)
                 if method_frame is not None:
                     try:
                         buffer = StringIO(body.decode("utf-8"))
                         df = cudf.io.read_json(buffer, orient='records', lines=True)
                         yield MessageMeta(df=df)
                     except Exception as ex:
-                        logger.exception("Error occurred converting RabbitMQ message to Dataframe: {}".format(ex))
+                        logger.exception("Error occurred converting RabbitMQ message to Dataframe: %s", ex)
```

Comment:
This is a pylint enforced change. At the time we introduced pylint to CI, @mdemoret-nv made the decision to go with the recommendation of:
1) Use %s style for logging as it avoids (potentially costly) string formatting when the log message won't be emitted (ex `logger.debug` when the logger is configured for info).
2) Use f-style for all other cases.

Reply:
The author of the reddit post bench-marked old-2.x style %s formatting, but didn't consider the special case of string formatting for log messages where string formatting is potentially avoided all together.

Reply:
How would string formatting be avoided altogether when the string contains `%s`, and why would we use an f-string at all if there's nothing to format?

Reply:
If you have:
```py
logger.exception(f"Error occurred converting RabbitMQ message to Dataframe: {ex}")
```
The interpreter will first format `ex` as a string (presumably by calling `ex.__str__()`), and then insert it into the string `"Error occurred converting RabbitMQ message to Dataframe: {ex}"`. Then it will call `logger.exception` passing in the resulting string.

On the other hand take:
```py
logger.exception("Error occurred converting RabbitMQ message to Dataframe: %s", ex)
```
The interpreter instead calls `logger.exception` passing in two arguments the string, and `ex`. If logging isn't enabled for the `ERROR` level, the method exits early without any string formatting being called.

Reply:
I guess that makes sense if you're optimizing for performance over readability in the fast case. :shrug: